src/core/machine-id-setup.c

```diff
@@ -190,7 +190,7 @@ static int generate_machine_id(char id[34], const char *root) {
         return 0;
 }
 
-int machine_id_setup(const char *root) {
+int machine_id_setup(const char *root, const char machine_id[34]) {
```

Comment:
Why not pass a `const char *` here and make sure the function has a sane fallback when `NULL` is given?